### PR TITLE
[MM-50269] Show app icons for disabled apps

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Fix a regression introduced in https://github.com/mattermost/mattermost-plugin-apps/pull/336 that blocked disabled apps from showing their icon.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50269

### CODE REVIEW

File: server/proxy/invoke_static.go [hunk 1]
```diff
@@ -35,13 +35,13 @@ func normalizeStaticPath(conf config.Config, appID apps.AppID, icon string) (str
...
 
 func (p *Proxy) InvokeGetStatic(r *incoming.Request, path string) (io.ReadCloser, int, error) {
-	app, err := p.getEnabledDestination(r)
+	app, err := p.GetApp(r)
 	if err != nil {
 		status := http.StatusInternalServerError
 		if errors.Is(err, utils.ErrNotFound) {
 			status = http.StatusNotFound
 		}
-		r.Log = r.Log.WithError(err)
```
Comment (hanzei) [inline server/proxy/invoke_static.go:44]: The error is logged by the caller anyway.

File: test/restapitest/call.go [hunk 2]
```diff
@@ -111,4 +115,20 @@ func testCalls(th *Helper) {
...
+			Values: map[string]interface{}{
+				builtin.FieldAppID: uninstallID,
+			},
+		}
+
+		cresp, _, err := th.Call(builtin.AppID, infoRequest)
+		require.NoError(th, err)
+		require.Equal(th, apps.CallResponseTypeError, cresp.Type)
+		require.Regexp(th, `user \w+ \(\w+\) is not a sysadmin: unauthorized`, cresp.Text)
+	})
```
Comment (hanzei) [inline test/restapitest/call.go:133]: The test belong here rather then in a mics file.

File: test/restapitest/static.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+		resp, err := th.UserClientPP.DoAPIGET(iconURL, "")
+		require.NoError(th, err)
+		require.NotNil(th, resp)
+		th.Cleanup(func() { _ = resp.Body.Close() })
+
+		data, err := io.ReadAll(resp.Body)
+		require.NoError(th, err)
+		require.Equal(th, iconPNG, data)
+	})
```
Comment (hanzei) [inline test/restapitest/static.go:68]: This is the real test that gets added.

**Review by javaguirre — APPROVED** [review]
Looks great, very nice testing coverage. 💯

File: server/proxy/invoke_static.go [hunk 4]
```diff
@@ -35,13 +35,13 @@ func normalizeStaticPath(conf config.Config, appID apps.AppID, icon string) (str
 }
 
 func (p *Proxy) InvokeGetStatic(r *incoming.Request, path string) (io.ReadCloser, int, error) {
-	app, err := p.getEnabledDestination(r)
+	app, err := p.GetApp(r)
```
Comment (levb) [inline server/proxy/invoke_static.go:38]: 👍 makes perfect sense

**Review by levb — APPROVED** [review]

**Pull request merged**